src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java

```diff
@@ -39,7 +40,11 @@ public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
   @Override
   public Uni<FindResponse> getDocuments(QueryExecutor queryExecutor) {
     QueryOuterClass.Query query = buildSelectQuery();
-    return findDocument(queryExecutor, query, pagingState, pageSize, readDocument, objectMapper);
+    if (ReadType.DOCUMENT == readType) {
```

Comment:
Does this work for all 3 types (if not, might want to use `switch`).
If it does, could simplify to

    return findDocument(queryExecutor, query, pagingState, pageSize, (ReadType.DOCUMENT == readType),
         objectMapper);

Reply:
Changed to code to switch